SplunkRumWorkspace/SplunkRum/SplunkRum/AppLifecycle.swift

```diff
@@ -94,3 +99,14 @@ func lifecycleEvent(_ event: String) {
 
     }
 }
+func invalidateSession(_ event: String) {
+    // 15 min inactivity then session time out
+    if event == "UIApplicationWillResignActiveNotification" {
+        sessionIdInActivityExpiration = Date().addingTimeInterval(TimeInterval(INACTIVITY_SESSION_TIMEOUT_SECONDS))
+    } else if event == "UIApplicationWillEnterForegroundNotification" {
+        if Date() > sessionIdInActivityExpiration { // expire 15 min
+           Is_New_SessionID_For_In_Activity = true
+            _  = getRumSessionId()
```

Comment:
Can't it just trigger new session id generation here instead?

Reply:
yes, it will create new session id after 15 mins of  inactivity.